### PR TITLE
pylint: accept specific 2 character names by default

### PR DESCRIPTION
**Pull request by villebro**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [x] Build / Development Environment
- [ ] Documentation

### SUMMARY
This PR adds a few commonly used two character variable names to the `pylint` config to avoid having to explicitly ignore them.

### REVIEWERS
@john-bodley

### CODE REVIEW

**Comment by john-bodley** [comment]
@villebro I think the `pylint` defaults (including three character variable names) are normally defined for a good reason. Why not just whitelist `id`, `df`, etc?

**Comment by etr2460** [comment]
Definitely agree with John's comment here, I'd rather we whitelist the specific 2 char variable names we think are acceptable.

Honestly, I wonder if we should change the variable name rules to match the other ones too

**Comment by villebro** [comment]
I'm fine with that approach, too. Personally I feel the three char minimum imposed by `pylint` is slightly overkill, but agree that excessive use of very short names is bad practice. If we go the whitelist route, I propose whitelisting `id` (class attribute), `df`, `fd` (form_data) and perhaps `ex` for exceptions.

**Comment by villebro** [comment]
@john-bodley @etr2460 this is ready for another round of review.

File: .pylintrc [hunk 1]
```diff
@@ -115,10 +115,10 @@ evaluation=10.0 - ((float(5 * error + warning + refactor + convention) / stateme
 [BASIC]
 
 # Good variable names which should always be accepted, separated by a comma
-good-names=i,j,k,ex,Run,_,d,e,v,o,l,x,ts,f
+good-names=_,d,df,e,ex,f,i,id,j,k,l,o,pk,Run,ts,v,x
```
Comment (dpgaspar) [inline .pylintrc:118]: nice!

**Review by dpgaspar — APPROVED** [review]
LGTM, we also have a bunch of pylint disables on the new API. Could be done on a different PR or address those here

**Comment by villebro** [comment]
@dpgaspar are you referring to `pk`? That's added here, but it seems a few new ones have popped up since last rebase, let me remove those. Are there any other ones we should address here, too?

File: .pylintrc [hunk 2]
```diff
@@ -115,10 +115,10 @@ evaluation=10.0 - ((float(5 * error + warning + refactor + convention) / stateme
...
 
 # Bad variable names which should always be refused, separated by a comma
-bad-names=foo,bar,baz,toto,tutu,tata,d,fd
+bad-names=d,fd,foo,bar,baz,toto,tutu,tata
```
Comment (john-bodley) [inline .pylintrc:121]: How is `d` both a good and bad name?

File: .pylintrc (hunk 1 above)
Comment (john-bodley) [reply .pylintrc:118]: I think `e` should be a bad name. We should favor `ex` for exceptions.

**Comment by villebro** [comment]
@john-bodley if you care to take another look, this turned into quite the refactor PR 😁

**Review by john-bodley — APPROVED** [review]
LGTM. Thanks for making the changes.

**Pull request merged**